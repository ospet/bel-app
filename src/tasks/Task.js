import React, {useState} from 'react';
import {Card} from 'antd';
import moment from 'moment';

const Task = ({
  title,
  description,
  dueDate,
  assignee
}) => {

  return !title ? null : (
    <>
      <Card data-testid="task-card" title={title} extra={moment(dueDate).format('YYYY-MM-DD')}>
        {description}
      </Card>
    </>
  )
}

export default Task;