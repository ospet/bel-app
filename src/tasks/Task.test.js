import React from 'react';
import { cleanup, render, screen } from '@testing-library/react';
import Task from './Task';

describe('Task', () => {
  afterEach(cleanup);

  it('renders null when no title', () => {
    const {asFragment} = render(<Task />);
    expect(asFragment(<Task />)).toMatchSnapshot();
  })

  it('renders basic task elements', () => {
    const {asFragment} = render(<Task
      title="TestTask"
      description="This is a Task under unit testing"
    />);

    expect(asFragment(<Task />)).toMatchSnapshot();
  });
});

