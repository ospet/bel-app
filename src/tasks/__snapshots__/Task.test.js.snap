// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task renders basic task elements 1`] = `
<DocumentFragment>
  <div
    class="ant-card ant-card-bordered"
    data-testid="task-card"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          TestTask
        </div>
        <div
          class="ant-card-extra"
        >
          2021-05-24
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      This is a Task under unit testing
    </div>
  </div>
</DocumentFragment>
`;

exports[`Task renders null when no title 1`] = `<DocumentFragment />`;
